frontend/src/metabase/dashboard/actions/writeback.ts

```diff
@@ -239,14 +243,44 @@ export type ExecuteRowActionPayload = {
   shouldToast?: boolean;
 };
 
+function hasDataFromExplicitAction(result: any) {
+  const isInsert = result["created-row"];
+  const isUpdate =
+    result["rows-affected"] > 0 || result["rows-updated"]?.[0] > 0;
+  const isDelete = result["rows-deleted"]?.[0] > 0;
+  return !isInsert && !isUpdate && !isDelete;
+}
+
+function getImplicitActionExecutionMessage(action: WritebackAction) {
+  if (action.slug === "insert") {
+    return t`Successfully saved`;
+  }
+  if (action.slug === "update") {
+    return t`Successfully updated`;
+  }
+  if (action.slug === "delete") {
+    return t`Successfully deleted`;
+  }
+  return t`Successfully run the action`;
+}
+
+function getActionExecutionMessage(action: WritebackAction, result: any) {
+  if (action.type === "implicit") {
+    return getImplicitActionExecutionMessage(action);
+  }
+  if (hasDataFromExplicitAction(result)) {
+    return t`Success! The action returned: ${JSON.stringify(result)}`;
+  }
+  return `${action.name} ${t`was run successfully`}`;
```

Comment:
is this any different than
```suggestion
  return t`${action.name} was run successfully`;
```
?